src/session/session_api.c

```diff
@@ -385,6 +400,9 @@ __session_open_cursor_int(WT_SESSION_IMPL *session, const char *uri,
 	if (*cursorp == NULL)
 		return (__wt_bad_object_type(session, uri));
 
+	if (owner != NULL)
+		F_CLR(*cursorp, WT_CURSTD_CACHEABLE);
+
```

Comment:
It's probably worth a comment inline to explain what's going on here: i.e., dependent cursors remain referenced by their owning cursor and are cached as part of caching the owning cursor.